File: 209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int res = Integer.MAX_VALUE;
        int n = nums.length;
        int sum = 0;

        for(int l = 0, r = 0; l < n && r < n; ++r){
            sum += nums[r];
            while(sum >= target){
                res = Math.min(res, r - l + 1);
                sum -= nums[l++];
            }
        }

        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: 209. Minimum Size Subarray Sum.cpp
#include"LeetCodeLibs.h"
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = INT_MAX;
        int n = nums.size();
        int sum = 0;
        
        for(int l = 0, r = 0; l < n && r < n; ++l){
            sum += nums[r];
            while(sum >= target){
                res = min(res, r - l + 1);
                sum -= nums[l++];
            }
        }

        return res == INT_MAX ? 0 : res;
    }
};